src/lib/mappingUtils.js

```diff
@@ -107,16 +107,80 @@ function mapPhase(procedure) {
   return outputs;
 }
 
+/**
+ * Takes a bundle of serviceRequests returned and returns an array of mappings of PlannedCourse data to be displayed in the table
+ * Based on https://build.fhir.org/ig/HL7/codex-radiation-therapy/StructureDefinition-codexrt-radiotherapy-planned-course.html
+ * @param {Object} procedure - A bundle of serviceRequests
+ * @returns {Object[]} Returns an array of objects with key/value pairs of data to display in the table
+ */
+function mapPlannedCourses(serviceRequests) {
+  const plannedCourses = fhirpath.evaluate(
+    serviceRequests,
+    "Bundle.entry.where(resource.code.coding.code = 'USCRS-33529').resource"
+  );
+  const outputs = [];
+  plannedCourses.forEach((plannedCourse) => {
+    const output = {};
+    output["Course Label"] = plannedCourse.identifier
+      ? plannedCourse.identifier[0].value
+      : "N/A";
+    output["Course Status"] = plannedCourse.status;
+    output["Request Intent"] = plannedCourse.intent;
+    output["Procedure Intent"] = fhirpath
+      .evaluate(
+        plannedCourse,
+        "ServiceRequest.extension.where(url = 'http://hl7.org/fhir/us/mcode/StructureDefinition/mcode-procedure-intent').valueCodeableConcept.single().coding"
+      )
+      .map((intent) => intent.display)[0];
+
+    output["Modalities"] = fhirpath
+      .evaluate(
+        plannedCourse,
+        "ServiceRequest.extension.where(url = 'http://hl7.org/fhir/us/mcode/StructureDefinition/mcode-radiotherapy-modality-and-technique').extension.where(url = 'http://hl7.org/fhir/us/mcode/StructureDefinition/mcode-radiotherapy-modality').valueCodeableConcept.coding"
+      )
+      .map((modality) => modality.display)[0];
+    output["Techniques"] = fhirpath
+      .evaluate(
+        plannedCourse,
+        "ServiceRequest.extension.where(url = 'http://hl7.org/fhir/us/mcode/StructureDefinition/mcode-radiotherapy-modality-and-technique').extension.where(url = 'http://hl7.org/fhir/us/mcode/StructureDefinition/mcode-radiotherapy-technique').valueCodeableConcept.coding"
+      )
+      .map((technique) => technique.display)
```

Comment:
This feels somewhat nitpicky but do we want some kind of null guard in the case that `technique.display` isn't present on this coding? (This is a demo app after all so feel free to say no, this hasn't been a problem thus far.)

Reply:
lol I should just get the display off the object in the fhir path